par_test.go

```diff
@@ -314,87 +313,143 @@ func BenchmarkReduce(b *testing.B) {
 	})
 }
 
-var result bool
-
 func BenchmarkAny(b *testing.B) {
-	rand.Seed(time.Now().Unix())
+	rand.Seed(1)
 	collections := CreateCollections(10000)
-	maxSum := CollectionsGetMaxSum(collections)
 
-	b.Run("serial", func(b *testing.B) {
-		var r bool
+	b.Run("serial with match", func(b *testing.B) {
 		for n := 0; n < b.N; n++ {
+			needle := collections[len(collections)*n/b.N].NumbersSum()
+			var result bool
 			for _, c := range collections {
-				if c.NumbersSum() == maxSum {
-					r = true
+				if c.NumbersSum() == needle {
+					result = true
 					break
 				}
 			}
+			_ = result
 		}
-		result = r
 	})
-	b.Run("parallel", func(b *testing.B) {
-		var r bool
+	b.Run("parallel with match", func(b *testing.B) {
 		for n := 0; n < b.N; n++ {
-			r = par.Any(collections, func(c Collection) bool {
-				return c.NumbersSum() == maxSum
+			needle := collections[len(collections)*n/b.N].NumbersSum()
+			_ = par.Any(collections, func(c Collection) bool {
+				return c.NumbersSum() == needle
+			})
+		}
+	})
+	b.Run("serial without match", func(b *testing.B) {
+		for n := 0; n < b.N; n++ {
+			needle := ^int(0)
+			var result bool
+			for _, c := range collections {
+				if c.NumbersSum() == needle {
+					result = true
+					break
+				}
+			}
+			_ = result
+		}
+	})
+	b.Run("parallel without match", func(b *testing.B) {
+		for n := 0; n < b.N; n++ {
+			needle := ^int(0)
+			_ = par.Any(collections, func(c Collection) bool {
```

Comment:
Not sure if this could result in the compiler optimising this out, as mentioned [here](https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go)

Similarly below.

Reply:
In the case of testing `t.*` it should not be possible in any case as the computation has observable side effects, but I added the same global assignments to those cases as well just to be 100% sure.